acceptancetests/clean_resources.py

```diff
@@ -27,28 +27,62 @@ def get_regions(args, env):
+                        'ap-south-1', 'ap-southeast-1', 'ap-southeast-2',
+                        'ap-northeast-1', 'ap-northeast-2',
+                        'sa-east-1']
+    logging.info('The target regions for cleaning job are \n{}'
+                 .format('\n'.join(selected_regions)))
     for region in selected_regions:
         with AWSAccount.from_boot_config(env, region=region) as substrate:
-            logging.info('Cleaning resources in {}.'.format(substrate.region))
-            all_groups = dict(substrate.iter_security_groups())
-            instance_groups = dict(substrate.iter_instance_security_groups())
-            non_instance_groups = dict((k, v) for k, v in all_groups.items()
-                                       if k not in instance_groups)
-            try:
-                unclean = substrate.delete_detached_interfaces(
-                    non_instance_groups.keys())
-                logging.info('Unable to delete {} groups'.format(len(unclean)))
-            except:
-                logging.info('Unable to delete non-instance groups {}'.format(non_instance_groups.keys()))
-            for group_id in unclean:
-                logging.debug('Cannot delete {}'.format(all_groups[group_id]))
-            for group_id in unclean:
-                non_instance_groups.pop(group_id, None)
-            try:
-                substrate.destroy_security_groups(non_instance_groups.values())
-            except:
-                logging.debug('Failed to delete groups {}'.format(non_instance_groups.values()))
+            if substrate is not None:
+                logging.info('Cleaning resources in {}.'
+                             .format(substrate.region))
+                all_groups = dict(substrate.iter_security_groups())
+                instance_groups = dict(substrate.iter_instance_security_groups())
+                logging.info('{} items found from instance groups on {} \n{}'
+                             .format(str(len(instance_groups)),
+                                     region,
+                                     '\n'.join(instance_groups)))
+                non_instance_groups = dict((k, v) for k, v in all_groups.items()
+                                           if k not in instance_groups)
+                logging.info('{} items found from non-instance groups on {} \n{}'
+                             .format(str(len(non_instance_groups)),
+                                     region,
+                                     '\n'.join(non_instance_groups)))
+                try:
+                    logging.info('{} detached interfaces will be deleted from {} \n{}'
+                                 .format(str(len(non_instance_groups.keys())),
+                                         region,
+                                         '\n'.join(non_instance_groups.keys())))
+                    unclean = substrate.delete_detached_interfaces(
+                        non_instance_groups.keys())
```

Comment:
If there is a instance group in the list that causes the exception to be raised we loose any information for any unclean or any follow up instances.

Perhaps a better mechanism would be to loop through each instance and call delete_detached_interfaces on it (accumulating any unclean).

This way we can log but ignore any exceptions raised by the call, ensuring that we get through the list of instances and try _all_ instances. Which means that any exception raised by delete_detached_interfaces won't leave us with an undefined unclean list.